src/components/DiscordImage.tsx

```diff
@@ -0,0 +1,24 @@
+"use client";
+
+import Image from "next/image";
+import useFocus from "~/hooks/use-focus";
+
+type DiscordImageProps = React.ComponentProps<typeof Image>;
+
+export default function DiscordImage(props: DiscordImageProps) {
+  const focused = useFocus();
+
+  const imageSrc =
+    typeof props.src === "string" &&
+    // ignore placeholders
+    !props.src.includes("/embed") &&
+    props.src.includes(".gif")
+      ? props.src.replace(
+          focused ? ".webp" : ".gif",
+          focused ? ".gif" : ".webp"
+        )
+      : props.src;
+
+  // eslint-disable-next-line
```

Comment:
Why is this required? What's the error message?

Reply:
It's the `alt` prop that was giving a warning.

Reply:
```suggestion
  // TODO: check why alt is not detected
  // eslint-disable-next-line
```